File: src/MeanscriptCpp/pub/MSOutputStream.cpp
#include "MS.h"
namespace meanscript {
using namespace meanscriptcore;

MSOutputStream::MSOutputStream ()
{
}

void MSOutputStream:: writeInt (int32_t i) 
{
	writeByte((uint8_t)((i>>24) & 0xff));
	writeByte((uint8_t)((i>>16) & 0xff));
	writeByte((uint8_t)((i>>8) & 0xff));
	writeByte((uint8_t)(i & 0xff));
}


} // namespace meanscript(core)

File: java/net/meanscript/MSOutputStream.java
package net.meanscript;
import net.meanscript.core.*;
import net.meanscript.java.*;
public abstract class MSOutputStream extends MC {

public MSOutputStream ()
{
}
public abstract void  writeByte (byte b) throws MException;
public abstract void  close () throws MException;

public void  writeInt (int i) throws MException
{
	writeByte((byte)((i>>24) & 0xff));
	writeByte((byte)((i>>16) & 0xff));
	writeByte((byte)((i>>8) & 0xff));
	writeByte((byte)(i & 0xff));
}


}
